src/Integration/LocalServices/ErrorListInfoBarController.cs

```diff
@@ -462,6 +491,190 @@ private ProjectViewModel FindProject(Uri serverUri, string projectKey)
                 return serverVM?.Projects.SingleOrDefault(p => ProjectInformation.KeyComparer.Equals(p.Key, projectKey));
             }
         }
+
+        internal /*testing purposes*/ class QualityProfileBackgroundProcessor : IDisposable
+        {
+            private readonly IHost host;
+            private bool isDisposed;
+
+            public QualityProfileBackgroundProcessor(IHost host)
+            {
+                if (host == null)
+                {
+                    throw new ArgumentNullException(nameof(host));
+                }
+
+                this.host = host;
+                this.TokenSource = new CancellationTokenSource();
+            }
+
+            internal /*for testing purposes*/ CancellationTokenSource TokenSource
+            {
+                get;
+            }
+
+            internal /*for testing purpose*/ System.Threading.Tasks.Task BackgroundTask
+            {
+                get;
+                private set;
+            }
+
+            public void QueueCheckIfUpdateIsRequired(Action updateAction)
+            {
+                if (updateAction == null)
+                {
+                    throw new ArgumentNullException(nameof(updateAction));
+                }
+
+                Debug.Assert(this.BackgroundTask == null, "Not expecting this method to be called more than once");
+
+                var projectSystem = this.host.GetService<IProjectSystemHelper>();
+                projectSystem.AssertLocalServiceIsNotNull();
```

Comment:
The AssertLocalServiceIsNotNull() is used in many places. It would be cleaner to have a method/extension method on the host instead e.g. _GetLocalService<>_ (assuming that local services should never be null)